src/server/routes/scanner.js

```diff
@@ -1,13 +1,229 @@
         res.render('scan-form', {
             description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum',
-            title: 'Analyze your website using sonar online'
+            title: 'Scanner'
         });
     });
 
-    app.post('/scanner', (req, res) => {
-        res.render('scan-result', { title: 'Results for your page | sonar' });
+    app.get('/scanner/api/:id', async (req, res) => {
+        const id = req.params.id;
+        let scanResult;
+
+        try {
+            scanResult = await queryResult(id);
+        } catch (error) {
+            return res.status(404);
+        }
+
+        const { categories } = processRuleResults(scanResult.rules);
+
+        return res.send({
+            categories,
+            status: scanResult.status,
+            time: calculateTimeDifference(scanResult.started, scanResult.status === jobStatus.finished ? scanResult.finished : void 0),
+            version: scanResult.sonarVersion
+        });
+    });
+
+    app.get('/scanner/:id', async (req, res) => {
+        const id = req.params.id;
+        let scanResult;
+
+        try {
+            scanResult = await queryResult(id);
+        } catch (error) {
+            return res.render('error', {
+                details: error.message,
+                heading: 'ERROR'
+            });
+        }
+
+        const includeIncompleteScan = scanResult.status === jobStatus.error;
+        const { categories, overallStatistics } = processRuleResults(scanResult.rules, includeIncompleteScan);
+        const renderOptions = {
+            categories,
+            id: scanResult.id,
+            layout,
+            overallStatistics,
+            permalink: `${sonarUrl}scanner/${scanResult.id}`,
+            status: scanResult.status,
+            time: calculateTimeDifference(scanResult.started, scanResult.status === jobStatus.finished ? scanResult.finished : void 0),
+            url: scanResult.url,
+            version: scanResult.sonarVersion
+        };
+
+        if (scanResult.status === jobStatus.error || scanResult.status === jobStatus.finished) {
+            renderOptions.isFinish = true;
+        }
+
+        res.render('scan-result', renderOptions);
+    });
+
+    app.post('/scanner', async (req, res) => {
+        if (!req.body || !req.body.url) {
+            return res.render('error', {
+                details: 'Please provide a url.',
+                heading: ''
+            });
+        }
+
+        let requestResult;
+
+        try {
+            requestResult = JSON.parse(await sendRequest(req.body.url));
```

Comment:
How long does it usually take to process this?
I'm thinking we should add `AppInsights` or similar to track the performance and duration of dependencies. I'll open a new issue for that.